### PR TITLE
fix: ElTable $index calculation issue (#2143)

### PR DESCRIPTION
**Pull request by msidolphin**
Please make sure these boxes are checked before submitting your PR, thank you!

bugfix #2187、#2143、#2208

* [x] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
* [x] Make sure you are merging your commits to `dev` branch.
* [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/table/__tests__/table.spec.ts [hunk 1]
```diff
@@ -2072,6 +2072,72 @@ describe('Table.vue', () => {
...
+      data() {
+        return {
+          testData: data,
+        }
+      },
+    })
+    await nextTick()
+    const rows = wrapper.findAll('.el-table__row')
+    rows.forEach((row, index) => {
+      const cell = row.findAll('td')[0]
```
Comment (HerringtonDarkholme) [inline packages/table/__tests__/table.spec.ts:2136]:
```suggestion
      const cell = row.find('td')
```

**Comment by HerringtonDarkholme** [comment]
Thanks for your contribution. But I guess this fix will break https://github.com/element-plus/element-plus/issues/1919

File: packages/table/src/table-body/index.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export default defineComponent({
             return acc.concat(
               this.wrappedRowRender(
                 row,
-                acc.filter(item => typeof item.key === 'number').length,
+                acc.length,
```
Comment (HerringtonDarkholme) [inline packages/table/src/table-body/index.ts:82]:
```suggestion
acc.filter(item => typeof item.key !== 'string' || !item.key.startsWith('expanded-row__')).length
```

**Review by HerringtonDarkholme — APPROVED** [review]
LGTM for now! Let's first merge it! ElTable still needs further test though. But this change is good to go IMHO.

File: packages/table/src/table-body/index.ts (hunk 2 above)
Comment (msidolphin) [reply packages/table/src/table-body/index.ts:82]: how about `acc.filter(row => row.key.toString().indexOf(EXPANDED_ROW_PREFIX) === -1).length` ？

> Thanks for your contribution. But I guess this fix will break #1919

I submitted another merge request (#2231)

**Pull request merged**